### PR TITLE
removed typedef uint32_richtig_huebsch

### PR DESCRIPTION
**Pull request by slizzered**
- replaced by `uint32_t` from `<boost/cstdint.hpp>`

### CODE REVIEW

File: src/include/mallocMC/mallocMC_utils.hpp [hunk 1]
```diff
@@ -100,7 +101,7 @@ namespace CUDA
 
 
 #define warp_serial                                    \
-  for (uint32_richtig_huebsch __mask = __ballot(1),                    \
+  for (uint32_t __mask = __ballot(1),                    \
```
Comment (ax3l) [inline src/include/mallocMC/mallocMC_utils.hpp]: I think the header signatures of these CUDA intrinsics are unfortunately [unsigned int](http://docs.nvidia.com/cuda/cuda-c-programming-guide/#warp-vote-functions).

I would therefore either write this explicitly or keep a typedef (with an other name) but not assuming `uint32_t`

**Comment by ax3l** [comment]
I would pass this to @psychocoderHPC if that is ok :)

**Comment by ax3l** [comment]
why do you need boost headers and not simple `<cstdint>` from the std libs?

File: src/include/mallocMC/mallocMC_utils.hpp (hunk 1 above)
Comment (slizzered) [inline src/include/mallocMC/mallocMC_utils.hpp]: wow, well spotted!
I will use the type directly, since this is the only line where this problem occurs.

**Comment by slizzered** [comment]
The rest of the project already uses `<boost/cstdint.hpp>`, so it was used for consistency.

AFAIK, `<cstdint>` is not available on Windows unless you use C++11, which was explicity not targeted at the time. So the Boost solution is mostly about compatibility between platforms and compiler versions.

**Comment by ax3l** [comment]
oh true, I didn't know [<cstdint>](http://www.cplusplus.com/reference/cstdint/) just became part of the stdlibs outside the *nix world in C++11

File: src/include/mallocMC/mallocMC_utils.hpp [hunk 2]
```diff
@@ -100,7 +101,7 @@ namespace CUDA
 
 
 #define warp_serial                                    \
-  for (uint32_richtig_huebsch __mask = __ballot(1),                    \
+  for (unsigned int __mask = __ballot(1),                    \
```
Comment (psychocoderHPC) [inline src/include/mallocMC/mallocMC_utils.hpp]: the backslash `\` is not aligned

**Pull request merged**